edge/pkg/common/util/config.go

```diff
@@ -1,103 +1,294 @@
 /*
-Copyright 2018 The KubeEdge Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-   http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
+CHANGELOG
+KubeEdge Authors:
+- This File is drived from github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/cert/cert.go
 */
 
 package util
 
 import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/rsa"
 	"crypto/x509"
 	"crypto/x509/pkix"
 	"encoding/pem"
+	"errors"
+	"fmt"
+	"math"
 	"math/big"
-	"os"
+	"net"
+	"path/filepath"
 	"time"
 
 	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
 	"k8s.io/klog/v2"
+	"k8s.io/kube-openapi/pkg/util/sets"
 )
 
-//GenerateTestCertificate generates fake certificates and stores them at the path specified.
-//It accepts 3 arguments path, certFileName and keyFileName
-// "path" is the directory path at which the directory is to be created,
-// "certFileName" & "keyFileName" refers to the name of the file to be created without the extension
-func GenerateTestCertificate(path string, certFileName string, keyFileName string) error {
-	template := &x509.Certificate{
-		IsCA:                  true,
-		BasicConstraintsValid: true,
-		SubjectKeyId:          []byte{1, 2, 3},
-		SerialNumber:          big.NewInt(1234),
+// PrepareTestCerts will generate rootCA and edge certificates in /tmp directory
+func PrepareTestCerts() error {
```

Comment:
Can [this function](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/servers/httpserver/server.go#L247) work also in this test case?
If it is possible, we can omit the description of karmada file.